flowr/src/main/java/com/fueled/flowr/Flowr.java

```diff
@@ -251,6 +288,46 @@ protected final String getTagPrefix() {
     }
 
     /**
+     * Parse the intent set by {@link #setDeepLinkContext(Intent)} and if this intent contains Deep Link info, update the {@link #currentFragment} and the Transaction data.
+     *
+     * @param data The Transaction data to extend if Deep link info are found in the {@link #deepLinkIntent}.
+     * @param <T>  The generic type for a valid Fragment.
+     */
+    @SuppressWarnings("unchecked")
+    private <T extends Fragment & FlowrFragment> void injectDeepLinkInfo(TransactionData<T> data) {
+        if (deepLinkIntent != null) {
+            try {
+                if (deepLinkHandler == null) {
+                    Constructor<? extends FlowrDeepLinkHandler> deepLinkCtor = findBindingConstructorForClass();
+                    if (deepLinkCtor != null) {
+                        deepLinkHandler = deepLinkCtor.newInstance();
+                    }
+                }
+                FlowrDeepLinkInfo info = deepLinkHandler.routeIntentToScreen(deepLinkIntent);
+                if (info != null) {
+                    data.setFragmentClass(info.fragment);
+                    Bundle dataArgs = data.getArgs();
+                    if (dataArgs != null) {
+                        data.getArgs().putAll(info.data);
+                    } else {
+                        data.setArgs(info.data);
+                    }
+                }
+            } catch (InstantiationException e) {
```

Comment:
@julien-fueled we can collapse this into one `catch` that accepts an `Exception` and use `Log.e` to log the exception.